euler-xyz/evk-periphery
# feat: pendle adapters

## PT-corn-SolvBTC.BBN-26DEC2024

- url: https://app.pendle.finance/trade/pools/0xeb4d3057738b9ed930f451be473c1ccc42988384
- available twap window: up to 18 minutes (1080)

## PT-LBTC-27MAR2025

- url: https://app.pendle.finance/trade/pools/0x70b70ac0445c3ef04e314dfda6caafd825428221
- available twap window: none

## PT-sUSDe-27MAR2025

- url: https://app.pendle.finance/trade/pools/0xcdd26eb5eb2ce0f203a84553853667ae69ca29ce
- available twap window: up to 30 minutes (1800)

## PT-staking-USDS-27MAR2025

- url: https://app.pendle.finance/trade/pools/0x977ebf77581f94de969349549ab2108a681e8f4c
- available twap window: none

## How to check observations buffer

- go to https://etherscan.io/address/0x9a9Fa8338dd5E5B2188006f1Cd2Ef26d921650C2#readProxyContract#F5
- call `getOracleState` with `pendleMarket` from the csv and duration in seconds
  - `increaseCardinalityRequired` should be false
  - `oldestObservationSatisfied` should be true
- these 2 conditions will be checked in `PendleOracle`'s constructor, so deployment will fail if the twap window is too large

## How to increase observations buffer

- check the observations buffer (above) for the desired duration and note the returned `cardinalityRequired`
- go to the `pendleMarket` contract on etherscan
- write `increaseObservationsCardinalityNext` with the `cardinalityRequired` value
